dronegen/tag.go

```diff
@@ -44,7 +45,12 @@ func tagCheckoutCommands(fips bool) []string {
 		// create necessary directories
 		`mkdir -p /go/cache /go/artifacts`,
 		// set version
-		`if [[ "${DRONE_TAG}" != "" ]]; then echo "${DRONE_TAG##v}" > /go/.version.txt; else egrep ^VERSION Makefile | cut -d= -f2 > /go/.version.txt; fi; cat /go/.version.txt`,
+		`VERSION=$(egrep ^VERSION Makefile | cut -d= -f2)
+if [ "$$VERSION" != "${DRONE_TAG##v}" ]; then
```

Comment:
Instead of using even more bash, can we please extend our [version check tool](https://github.com/gravitational/teleport/tree/v9.0.1/build.assets/tooling/cmd/check) to support this and use it instead? It's already used in some other places e.g. [checking for pre-release](https://github.com/gravitational/teleport/blob/v9.0.1/.drone.yml#L4894).

Reply:
`cmd/check` does not seem to exist in branch/v7, and does not seem to be fully backported (up-to-date) in branch/v8. We could try to backport it everywhere, perhaps.

Reply:
> `cmd/check` does not seem to exist in branch/v7, and does not seem to be fully backported (up-to-date) in branch/v8. We could try to backport it everywhere, perhaps.

I suspect https://github.com/gravitational/teleport/pull/10436 is where the port was not fully backported.

> I chose not to backport the docker image publishing fixes (for https://github.com/gravitational/teleport/issues/9494), as that drone pipeline is only ever run off master.

If it helps these efforts, I can backport the rest of https://github.com/gravitational/teleport/pull/10295 to v8 and v7.  Let me know.